src/lib.rs

```diff
@@ -248,12 +248,24 @@ pub enum Error {
     MissingHash(sha256::Hash),
     /// Could not satisfy a script (fragment) because of a missing signature
     MissingSig(bitcoin::PublicKey),
-    /// Could not satisfy, locktime not met
-    LocktimeNotMet(u32),
+    /// Could not evaluate a script (fragment) because of error in script execution
+    CouldnotEvaluate(String),
+    /// Could not satisfy, relative locktime not met
+    RelativeLocktimeNotMet(u32),
+    /// Could not satisfy, absolute locktime not met
+    AbsoluteLocktimeNotMet(u32),
     /// General failure to satisfy
     CouldNotSatisfy,
     /// Typechecking failed
     TypeCheck(String),
+    /// Could not evaluate a script (fragment) because of a invalid signature
+    VerifySigFail(bitcoin::PublicKey, secp256k1::Signature, secp256k1::Message),
+    /// Could not evaluate a script (fragment) because of a invalid hashlock
+    VerifyHashFail(sha256::Hash),
```

Comment:
All of the above errors (`BadDescriptor` and `Secp` are okay, though `BadDescriptor` should be split into multiple errors) should be wrapped up in a single `interpreter::Error` type)